### PR TITLE
feat: add support for account id

### PR DESCRIPTION
**Pull request by ianaya89**
## Affected Components
* [x] New Features
* [ ] Bug Fixing
* [ ] Types
* [ ] Tests
* [x] Docs
* [ ] Other

## Style
* [x] Go code is formatted with `go fmt`

## Notes for the Reviewer
Add support for setting Account ID (only for User API Keys)

> Resolves https://github.com/checkly/checkly-go-sdk/issues/51

### CODE REVIEW

File: checkly.go [hunk 1]
```diff
@@ -45,6 +45,11 @@ func NewClient(
 	return c
 }
 
+// Set checkly account id (required when using User API keys)
+func (c *client) SetAccountId(ID string) {
```
Comment (danielpaulus) [inline checkly.go:49]: checkout godoc guidelines https://go.dev/blog/godoc
It should be sth. more like 
```SetAccountId sets ID on a client which is required when using User API keys.```

File: checkly.go [hunk 2]
```diff
@@ -810,11 +815,20 @@ func (c *client) apiCall(
...
 	if err != nil {
 		return 0, "", fmt.Errorf("failed to create HTTP request: %v", err)
 	}
+
 	req.Header.Add("Authorization", "Bearer "+c.apiKey)
 	req.Header.Add("content-type", "application/json")
+
+	if strings.HasPrefix(c.apiKey, "cu") && c.accountId == "" {
+		return 0, "", fmt.Errorf("Missing Checkly Account ID (required when using User API Keys)")
+	} else if c.accountId != "" {
```
Comment (danielpaulus) [inline checkly.go:828]: I think `else` is not needed here, it only makes the code harder to grasp

**Review by danielpaulus — CHANGES_REQUESTED** [review]
Looks good in general, just two minor nitpicks.

On a side note: if you want to be really fancy in future releases, I'd recommend implementing functional config:
example: https://gist.github.com/joyrexus/33d2e059ab5812ee7f0f

It is a super nice pattern.

**Comment by ianaya89** [comment]
@danielpaulus I like the functional config pattern! It is something I really want to add here but I don't want to introduce breaking changes in the version.
Will add a ticket to tackle that in v2.

**Review by danielpaulus — APPROVED** [review]

**Pull request merged**